components/backup/src/endpoint.rs

```diff
@@ -151,6 +152,118 @@ pub struct BackupRange {
     cf: CfName,
 }
 
+/// The generic saveable writer. for generic `InMemBackupFiles`.
+/// Maybe what we really need is make Writer a trait...
+enum KvWriter {
+    TiDB(BackupWriter),
+    Raw(BackupRawKVWriter),
+}
+
+impl KvWriter {
+    async fn save(self, storage: &dyn ExternalStorage) -> Result<Vec<File>> {
+        match self {
+            Self::TiDB(writer) => writer.save(storage).await,
+            Self::Raw(writer) => writer.save(storage).await,
+        }
+    }
+
+    fn need_flush_keys(&self) -> bool {
+        match self {
+            Self::TiDB(writer) => writer.need_flush_keys(),
+            Self::Raw(_) => true,
+        }
+    }
+}
+
+struct InMemBackupFiles {
+    files: KvWriter,
+    start_key: Vec<u8>,
+    end_key: Vec<u8>,
+    start_version: TimeStamp,
+    end_version: TimeStamp,
+    region: Region,
+}
+
+impl std::fmt::Debug for InMemBackupFiles {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("InMemBackupFiles")
```

Comment:
should just implement a dummy `Debug` for `KvWriter` and `#[derive(Debug)]` for `InMemBackupFiles`.